.github/workflows/format-checker.yml

```diff
@@ -0,0 +1,28 @@
+name: Check Formatting
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Install dependencies from APT
+      run: |
+        sudo apt install git clang-format
```

Comment:
Should probably specify -y to auto confirm any prompt from apt.

Reply:
Oh good catch I'll add that